src/main/java/gregtech/api/util/GTUtility.java

```diff
@@ -249,6 +249,41 @@ public static boolean mergeItemStack(ItemStack itemStack, List<Slot> slots, bool
         return merged;
     }
 
+    /**
+     * Attempts to merge given ItemStack with ItemStacks in list supplied
+     * growing up to their max stack size
+     *
+     * @param stackToAdd item stack to merge.
+     * @return a list of stacks, with optimized stack sizes
+     */
+
+    public static List<ItemStack> addStackToItemStackList(ItemStack stackToAdd, List<ItemStack> itemStackList){
+        if (!itemStackList.isEmpty()) {
+            for (ItemStack stackInList : itemStackList) {
+                if (ItemStackHashStrategy.comparingAllButCount().equals(stackInList, stackToAdd)) {
```

Comment:
There could be a short circuit check before this comparison call, checking if `stackToAdd.isEmpty()`.

Alternatively, as I mentioned later down in this method, the break should be moved to after `stackToAdd` gets set to the empty itemstack, which would also remove the need for this short circuit.